### PR TITLE
[BEAM-121] Add DisplayData for composite IO transforms

### PR DESCRIPTION
**Pull request by swegner**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by swegner** [comment]
R: @dhalperi

File: sdks/java/core/src/main/java/org/apache/beam/sdk/io/AvroIO.java [hunk 1]
```diff
@@ -327,6 +328,14 @@
       }
 
       @Override
+      public void populateDisplayData(DisplayData.Builder builder) {
+        super.populateDisplayData(builder);
+        builder
+          .addIfNotNull("filePattern", filepattern)
```
Comment (dhalperi) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/io/AvroIO.java]: is filepattern not in the `super`?

Comment (dhalperi) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/io/AvroIO.java]: oh never mind I'm forgetting our discussion. And no, it's not, it's in the FileBasedSource inside.

File: sdks/java/core/src/main/java/org/apache/beam/sdk/io/BigQueryIO.java [hunk 2]
```diff
@@ -1048,6 +1062,22 @@ public PDone apply(PCollection<TableRow> input) {
...
+
+        if (tableRefFunction != null) {
+          builder.add("tableFn", tableRefFunction.getClass());
+        }
+
+        builder
+            .add("createDisposition", createDisposition.toString())
+            .add("writeDisposition", writeDisposition.toString())
+            .addIfNotNull("table", jsonTableRef)
+            .addIfNotNull("schema", jsonSchema)
```
Comment (dhalperi) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/io/BigQueryIO.java]: if ordering is or ever will be meaningful, I'd move table/schema to be the first two here.

**Comment by swegner** [comment]
I've addressed all feedback so far. Please take another look. @dhalperi

**Comment by swegner** [comment]
Backported via https://github.com/GoogleCloudPlatform/DataflowJavaSDK/pull/218

**Pull request closed without merging**